backend/src/openai.ts

```diff
@@ -141,14 +141,14 @@ function isChatGptFunction(functionName: string) {
 async function handleAskQuestionFunction(
 	functionCallArgs: string | undefined,
 	currentLevel: LEVEL_NAMES,
-	defences: Defence[]
+	qaLlmDefence?: QaLlmDefence
 ) {
 	if (functionCallArgs) {
 		const params = JSON.parse(functionCallArgs) as FunctionAskQuestionParams;
 		console.debug(`Asking question: ${params.question}`);
 		// if asking a question, call the queryDocuments
-		const configQAPrompt = isDefenceActive(DEFENCE_ID.QA_LLM, defences)
-			? getQAPromptFromConfig(defences)
+		const configQAPrompt = qaLlmDefence?.isActive
+			? getQAPromptFromConfig([qaLlmDefence])
```

Comment:
If you don't change Defence to that union type, you might need to do `[qaLlmDefence as Defence]` here.

Reply:
seems to work fine before and after changing defence from that union type